### DISCUSSION TITLE
no response when defining the Objective

### DISCUSSION THREAD

**Discussion by Hurricane-k** [yalmip/YALMIP#1214] [Problems using YALMIP]
first, I define two `spdvar`, the first one is a matrix of size (30,8), and another is a matrix of size (10,1)
```
co_srd = sdpvar(30,8); # the size of co_spd is (30,8)
co_spd = sdpvar(10,1);
```
and then I defined the Objective
```
# sample_values is a known matrix of size (30,1), A is a known matrix of size (8,10), 
# so the temp_obj is a matrix of size (30,1) containing  spdvar 
temp_obj = sample_values-co_srd*A*co_spd;
```
but when I run the next command, **there is no response**, just not warning, no error, have no idea how to deal with it
```
Objective = temp_obj'*temp_obj; 
```

**Comment by Hurricane-k** [comment]
I find if decrease the dimension of the spdvar, `co_srd` decrease to size of (16,8), the Objective can be done. so the higher dimension makes a huge gap?

**Comment by Hurricane-k** [comment]
i GOT the problem, Matlab console shows
```
Error using repmat Requested 65920x66186 (32.5GB) array exceeds maximum array size preference.
Creation of arrays greater than this limit may take a long time and cause MATLAB to become unresponsive. 
See array size limit or preference panel for more information.
```

**Comment by johanlofberg** [comment]
The number of terms there will be in that quartic expression will just be outside the capabilites/scope of YALMIP. Even it did manage to  create that objective, what solver did you hope would be capable of solving that.

One way to reduce the complexity of the expression is to add a new variable e, add the nonconvex quadratic constraints e == temp_obj and then use sparse convex quadratic objective e'*e

**Reply by Hurricane-k** [reply]
I didn't set the specific solver in `sdpsettings`, just directly `sol = optimize(Constraints, Objective)`, default everything in 'sdpsettings', coz my goal is to solve the problem, not pick the best-performing solver.
 
By the way, actually my original `spdvar` are a matrix of size (262144, 8) and another matrix of size (10,1), i dont know how to create the objective, can avoid being out of memory. my laptop only has 16GB RAM 🤣

**Reply by johanlofberg** [reply]
Well, regardless you must have an idea of what solver you intended to use, as this is a massively large, and extremely hard problem. Sounds like you are too optimistic.

To have any kind of hope you will have to write your own layer to call a solver, with efficient computation of gradients and objectives, and then hope that the solver has any kind of progress on that nonconvex problem.

**Reply by Hurricane-k** [reply]
thank you for your advise, My major is engineering, but mathematics, have to admit that my maths foundation needs to improve. not defining solver still can run. 
For my current problem, my understand is that non-linear objective (highest is four-order), is nonconvex, simple linear constriants. coding base layer seems to be too hard and impractical, could you give me other advice than coding base layer

**Reply by johanlofberg** [reply]
coding base layer?

**Reply by Hurricane-k** [reply]
sry, I misunderstood, you said write my own layer to call a solver, my understand is that i have to wrtie a solver for my problem. i have no idea what is my own layer to call a solver

**Reply by johanlofberg** [reply]
yes, you either have to write your own solver, or interface a solver directly to allow you do efficiently compute the objective and gradient without any symbolic overhead from yalmip (but then still remember that all you can hope for is that you can only, at best, a locally optimal solution)

**Reply by Hurricane-k** [reply]
really apprecitate it, thanks for your time and suggestion
